src/core/server/logging/logging_system.ts

```diff
@@ -154,17 +161,21 @@ export class LoggingSystem implements LoggerFactory {
     return this.getLoggerConfigByContext(config, LoggingConfig.getParentLoggerContext(context));
   }
 
-  private applyBaseConfig(newBaseConfig: LoggingConfig) {
+  private async applyBaseConfig(newBaseConfig: LoggingConfig) {
     const computedConfig = [...this.contextConfigs.values()].reduce(
       (baseConfig, contextConfig) => baseConfig.extend(contextConfig),
       newBaseConfig
     );
 
+    // reconfigure all the loggers without configuration to have them use the buffer
+    // appender while we are awaiting for the appenders to be disposed.
+    for (const [loggerKey, loggerAdapter] of this.loggers) {
+      loggerAdapter.updateLogger(this.createLogger(loggerKey, undefined));
+    }
+
     // Appenders must be reset, so we first dispose of the current ones, then
     // build up a new set of appenders.
-    for (const appender of this.appenders.values()) {
-      appender.dispose();
-    }
+    await Promise.all([...this.appenders.values()].map((a) => a.dispose()));
```

Comment:
Small change, heavy implications. 

We were not awaiting for the previous appenders to dispose before creating the new ones (which was a mistake, we are actually doing it anywhere else where we dispose appenders, such as `LoggingSystem.stop`.

Effectively, if this had low to no impact with previous appenders, it could cause race conditions with `rolling-file`, as if a rolling was in progress when going through `applyBaseConfig `, we were not awaiting for it to complete, which causes the newly created appender to also start a rollover, resulting in a race on the files removal/renaming.

If the change is trivial, `applyBaseConfig` is now asynchronous, which required to adapt the consumer chain of this API.